redis/src/cluster_async/connections_container.rs

```diff
@@ -186,83 +175,61 @@ where
+            .map(move |(address, node)| (address.clone(), node.user_connection.clone()))
     }
 
     pub(crate) fn all_primary_connections(
         &self,
-    ) -> impl Iterator<Item = ConnectionAndIdentifier<Connection>> + '_ {
+    ) -> impl Iterator<Item = ConnectionAndAddress<Connection>> + '_ {
         self.slot_map
             .addresses_for_all_primaries()
             .into_iter()
             .flat_map(|addr| self.connection_for_address(addr))
     }
 
-    fn node_for_identifier(&self, identifier: &Identifier) -> Option<ClusterNode<Connection>> {
-        let node = self.connection_map.get(identifier)?;
-        Some(node.clone())
-    }
-
     pub(crate) fn node_for_address(&self, address: &str) -> Option<ClusterNode<Connection>> {
-        let identifier = Identifier(address.into());
-        let node = self.node_for_identifier(&identifier)?;
-        Some(node.clone())
-    }
-
-    pub(crate) fn connection_for_identifier(&self, identifier: &Identifier) -> Option<Connection> {
-        let node = self.connection_map.get(identifier)?;
-        Some(node.user_connection.clone())
+        self.connection_map.get(address).cloned()
     }
 
     pub(crate) fn connection_for_address(
         &self,
         address: &str,
-    ) -> Option<ConnectionAndIdentifier<Connection>> {
-        let identifier = Identifier(address.into());
-        let connection = self.connection_for_identifier(&identifier)?;
-        Some((identifier, connection))
-    }
-
-    pub(crate) fn address_for_identifier<'a, 'b: 'a>(
-        &'a self,
-        identifier: &'a Identifier,
-    ) -> Option<ArcStr> {
-        Some(identifier.0.clone())
+    ) -> Option<ConnectionAndAddress<Connection>> {
+        self.connection_map
+            .get_key_value(address)
+            .map(|(address, conn)| (address.clone(), conn.user_connection.clone()))
     }
 
     pub(crate) fn random_connections(
         &self,
         amount: usize,
         conn_type: ConnectionType,
-    ) -> impl Iterator<Item = ConnectionAndIdentifier<Connection>> + '_ {
+    ) -> impl Iterator<Item = ConnectionAndAddress<Connection>> + '_ {
         self.connection_map
             .iter()
             .choose_multiple(&mut rand::thread_rng(), amount)
             .into_iter()
-            .map(move |(identifier, node)| {
+            .map(move |(address, node)| {
                 let conn = node.get_connection(&conn_type);
-                (identifier.clone(), conn)
+                (address.clone(), conn)
             })
     }
 
     pub(crate) fn replace_or_add_connection_for_address(
         &mut self,
         address: impl Into<ArcStr>,
         node: ClusterNode<Connection>,
-    ) -> Identifier {
-        let identifier = Identifier(address.into());
-        self.connection_map.insert(identifier.clone(), node);
-        identifier
+    ) -> ArcStr {
+        let address = address.into();
+        self.connection_map.insert(address.clone(), node);
+        address
```

Comment:
Do we still need to return the address even though the caller provided it?

Reply:
nope!

Reply:
sorry - while not necessary, it is convenient, since the input isn't ArcStr but ` impl Into<ArcStr>` - we get the output address in a way which is cheaply clonable.
This is used in the cluster connection, so keeping for now.